src/Titon/Console/Console.hh

```diff
@@ -0,0 +1,161 @@
+<?hh // strict
+/**
+ * @copyright   2010-2015, The Titon Project
+ * @license     http://opensource.org/licenses/bsd-license.php
+ * @link        http://titon.io
+ */
+
+namespace Titon\Console;
+
+use Titon\Console\Input;
+use Titon\Console\Output;
+use Titon\Console\InputDefinition\Flag;
+use ReflectionClass;
+
+/**
+ * The `Console` class bootstraps and handles Input and Output to process and
+ * run necessary commands.
+ *
+ * @package Titon\Console
+ */
+class Console {
+
+    /**
+     * The `Command` object to run
+     *
+     * @var \Titon\Console\Command|null
+     */
+    protected ?Command $command;
+
+    /**
+     * The `Input` object used to retrieve parsed parameters and commands.
+     *
+     * @var \Titon\Console\Input
+     */
+    protected Input $input;
+
+    /**
+     * The `Output` object used to send response data to the user.
+     *
+     * @var \Titon\Console\Output
+     */
+    protected Output $output;
+
+    /**
+     * Construct a new `Console` application.
+     */
+    public function __construct(?Input $input = null, ?Output $output = null) {
+        if (is_null($input)) {
+            $input = new Input();
+        }
+
+        if (is_null($output)) {
+            $output = new Output();
+        }
+
+        $this->input = $input;
+        $this->output = $output;
+    }
+
+    /**
+     * Add a `Command` to the application to be parsed by the `Input`.
+     *
+     * @param Command $command  The `Command` object to add
+     *
+     * @return $this
+     */
+    public function addCommand(string $command): this {
```

Comment:
The only necessity of having it in the constructor is that we allow the input/output class variable to be a nullable type (since it is not always set according to the type checker), so every time we call an alias interacting with the input/output, we will have to set to a local variable and invariant it.